common/src/main/java/com/wynntils/mc/mixin/ClientPacketListenerMixin.java

```diff
@@ -206,19 +264,79 @@ private void setSubtitleTextPre(ClientboundSetSubtitleTextPacket packet, Callbac
         }
     }
 
-    @Redirect(
-            method = "handleChat(Lnet/minecraft/network/protocol/game/ClientboundChatPacket;)V",
+    @Inject(
+            method = "handlePlayerChat",
+            at =
+                    @At(
+                            value = "INVOKE",
+                            target =
+                                    "Lnet/minecraft/client/multiplayer/chat/ChatListener;handlePlayerChatMessage(Lnet/minecraft/network/chat/PlayerChatMessage;Lcom/mojang/authlib/GameProfile;Lnet/minecraft/network/chat/ChatType$Bound;)V"),
+            cancellable = true)
+    private void handlePlayerChat(ClientboundPlayerChatPacket packet, CallbackInfo ci) {
+        if (!isRenderThread()) return;
+        ChatPacketReceivedEvent result =
+                EventFactory.onChatReceived(com.wynntils.mc.objects.ChatType.CHAT, packet.unsignedContent());
```

Comment:
So this entire mixin is not needed, since Wynncraft does not send ClientboundPlayerChatPacket in 1.19? Can we remove it if that is the case?

Reply:
I would keep it just so we can process it in case Bukkit decides to update and use it for some reason? I know it is quite large, but I would rather have the ability to do this.

Reply:
ok